### DISCUSSION TITLE
ID not present when using Authentik as a provider

### DISCUSSION THREAD

**Discussion by chpiotr06** [nextauthjs/next-auth#8274] [Help]
Hello! I am creating dashboard app in next js utilizing app directory and am unable to determine why id is not present in session object provided by getServerSession function. 

I am using next@13.4.12 and next-auth@4.22.3.

```ts
// app/api/auth/[...nextauth]/route.ts
import NextAuth from "next-auth/next";
import AuthentikProvider from "next-auth/providers/authentik";

export const authOptions = {
    providers: [AuthentikProvider({
        clientId: process.env.AUTHENTIK_CLIENT_ID,
        clientSecret: process.env.AUTHENTIK_CLIENT_SECRET,
        issuer: process.env.AUTHENTIK_ISSUER,
    })],
}

const handler = NextAuth(
    authOptions
)

export {handler as GET, handler as POST}
```
I would expect that calling this api route logs profile structure presented here:  [url from authentik provider documentation](https://github.com/nextauthjs/next-auth/blob/main/packages/next-auth/src/providers/authentik.ts), however id is missing. 

```ts
//app/api/test/route.ts

export async function GET(){
    const session = await getServerSession(authOptions)
    console.log(session)

    return NextResponse.json({})
}
```
Middleware just exports default next-auth middleware.

**Comment by UnsolvedCypher** [comment]
You may need to override the [session callback](https://next-auth.js.org/configuration/callbacks#session-callback). By default, the session contains only a few properties, others like the id must be forwarded manually in that callback. The callbacks would go in your authOptions object, see [here](https://next-auth.js.org/configuration/options#callbacks).

**Comment by itsdapi** [comment]
Hi, I have been digging into this issue for a while. For me, nextauth seems not to recognize the same user and try to create a new account each time logging in. Later on, I found out the ID is not present in the profile object. All I found to identify the user is using `sub` property. So what I do is just use the profile function to modify it.

> make sure to set the provider `subject mode` to `Based on hash id` or something else won't change unlike username or email.

```typescript
AuthentikProvider({
      clientId: config.auth.autho.id,
      clientSecret: config.auth.autho.secret,
      issuer: config.auth.autho.issuer,
      profile(profile) {
        return {
          ...profile,
          id: profile.sub,
        }
      }
    })
```

I am not quite familiar with oidc and Oauth. If anyone knows a better solution please hit me.
